packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -94,6 +100,7 @@ export default createRule<
   defaultOptions: [
     {
       allowPotentiallyUnsafeFixesThatModifyTheReturnTypeIKnowWhatImDoing: false,
+      allowSuggestingOnIfStatements: false,
```

Comment:
For this major version (v8) I think it makes sense to leave this off by default. But in general, this is a good change that makes the rule more comprehensive. I suspect we'll want to enable it by default in the next major version that changes defaults (presumably, v10).

_"allow suggesting X"_ is also a bit of a double negative. Most `allow*` options make rules more _lenient_.

Suggestion: let's have this be `true` by default, with the intent to make it more strict later?

```suggestion
      allowIfStatements: true,
```

(this'll also need a docs update)

Reply:
The rule as it is (when `requireNullish` is false/default) is not completely type-safe, and I was wondering if it should continue this way with only suggesting instead of fixing ([example for both](https://deploy-preview-10137--typescript-eslint.netlify.app/play/#ts=5.5.2&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3gMyUgXDlBGggD7wDe8ARrIQGpICWwAYgK4pgbOrwBfANwAoEc2zwAFLiQA6OjACUVEfBx4FsKUtECxsrTHgAyExvmKA-HJ1CgA&eslintrc=N4KABGBECmAeAu0B2ATAzpAXGA2uCYA9IVAA4A2ArgOYCWSmAAvAJ6nRoDGATrafAFoO5evELdonAPYBbGchTQUAiQEdKtXkmoDW7AZwAWkgNb1qkfAF0ANPkjdK5DljCgCUZmw48%2Bg4aKEpBIAZtDcAlL8tFJIAIbkBoZx9K44MNzcUtyQNm74HpAJ5FIA7gDKNNQc8OYA8kgAkiHl8HGI8kjwGNjwjtB2HhAO0Oqa0AByTiJohq4hCWjQBWAAvlb4qyCrQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eFYDAruuGAL4g9A&tokens=false)) I also messed it up and made it an autofix here, I could use some advice on this.

I also wanted to convey that the rule doesn't apply to any optional inside if, but only to single statement call expressions but it's too much.

Naming things is hard :)

Reply:
Haha, it really is...

I think the new if statements should follow the same logic as existing code. `if (abc) { abc.def }` to me is mentally the same model as `abc && abc.def`. So as long as they're consistent, and the new option just stops the rule from reporting on the `if` case, I'm happy.